.github/workflows/build.yaml

```diff
@@ -0,0 +1,78 @@
+name: Build
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - '*'
+  pull_request:
+    branches:
+    - master
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  # Increment this key to invalidate the cache.
+  CACHE_KEY: 0
+
+jobs:
+  all:
+    name: All
+
+    strategy:
+      matrix:
+        os:
+        - macos-latest
+        - ubuntu-latest
+        - windows-latest
+
+    runs-on: ${{matrix.os}}
+
+    env:
+      RUSTFLAGS: "-D warnings"
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # An issue with BSD Tar causes sporadic failures on macOS.
+    # c.f https://github.com/actions/cache/issues/403
+    - name: Install GNU Tar
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew install gnu-tar
+        echo /usr/local/opt/gnu-tar/libexec/gnubin > $GITHUB_PATH
+
+    - name: Cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/registry
+          ~/.cargo/git
+          target
+        key: cargo-${{ env.CACHE_KEY }}-${{ runner.os }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Set up the cache per OS, and invalidate it if any Cargo.lock file changes. Invalidating it on any Cargo.lock file change seems aggressive to me, but it's what the documentation suggests.